### PR TITLE
MAC-166-Replace calls to Reservations API

### PR DESCRIPTION
**Pull request by ratheeshri1984**

### CODE REVIEW

File: src/SFA.DAS.EmployerAccounts/DependencyResolution/ConfigurationRegistry.cs [hunk 1]
```diff
@@ -20,7 +19,7 @@ public ConfigurationRegistry()
             For<EmployerFinanceConfiguration>().Use(c => c.GetInstance<IAutoConfigurationService>().Get<EmployerFinanceConfiguration>(ConfigurationKeys.EmployerFinance)).Singleton();
             For<EmployerAccountsReadStoreConfiguration>().Use(c => c.GetInstance<IAutoConfigurationService>().Get<EmployerAccountsReadStoreConfiguration>(ConfigurationKeys.EmployerAccountsReadStore)).Singleton();
             For<ReferenceDataApiClientConfiguration>().Use(c => c.GetInstance<IAutoConfigurationService>().Get<ReferenceDataApiClientConfiguration>(ConfigurationKeys.ReferenceDataApiClient)).Singleton();
-            For<ReservationsClientApiConfiguration>().Use(c => c.GetInstance<IAutoConfigurationService>().Get<ReservationsClientApiConfiguration>(ConfigurationKeys.ReservationsClientApiConfiguration)).Singleton();
+            For<EmployerAccountsOuterApiConfiguration>().Use(c => c.GetInstance<IAutoConfigurationService>().Get<EmployerAccountsOuterApiConfiguration>(ConfigurationKeys.ReservationsClientApiConfiguration)).Singleton();
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerAccounts/DependencyResolution/ConfigurationRegistry.cs:22]: the key being passed in here to get the config section is incorrect. `EmployerAccountsOuterApiConfiguration` should come from `c => c.GetInstance<EmployerAccountsConfiguration>().EmployerAccountsOuterApiConfiguration`

File: src/SFA.DAS.EmployerAccounts/Interfaces/IReservationsClientApiConfiguration.cs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 namespace SFA.DAS.EmployerAccounts.Interfaces
 {
-    public interface IReservationsClientApiConfiguration : IAzureActiveDirectoryClientConfiguration
+    public interface IReservationsClientApiConfiguration : IManagedIdentityClientConfiguration
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerAccounts/Interfaces/IReservationsClientApiConfiguration.cs:5]: this is never used. So delete?

File: src/SFA.DAS.EmployerAccounts/DependencyResolution/ConfigurationRegistry.cs (hunk 1 above)
Comment (cofaulco) [reply src/SFA.DAS.EmployerAccounts/DependencyResolution/ConfigurationRegistry.cs:22]: in fact, the configuration is registered in the newly created `EmployerAccountsOuterApiRegistry` so this line should be removed.

Comment (ratheeshri1984) [reply src/SFA.DAS.EmployerAccounts/DependencyResolution/ConfigurationRegistry.cs]: Done

File: src/SFA.DAS.EmployerAccounts/Interfaces/IReservationsClientApiConfiguration.cs (hunk 2 above)
Comment (ratheeshri1984) [reply src/SFA.DAS.EmployerAccounts/Interfaces/IReservationsClientApiConfiguration.cs]: Done

**Review by cofaulco — APPROVED** [review]

File: src/SFA.DAS.EmployerAccounts/DependencyResolution/ConfigurationRegistry.cs (hunk 1 above)
Comment (ratheeshri1984) [reply src/SFA.DAS.EmployerAccounts/DependencyResolution/ConfigurationRegistry.cs]: done

File: src/SFA.DAS.EmployerAccounts/Interfaces/IReservationsClientApiConfiguration.cs (hunk 2 above)
Comment (ratheeshri1984) [reply src/SFA.DAS.EmployerAccounts/Interfaces/IReservationsClientApiConfiguration.cs]: done

**Pull request merged**